js/msp.js

```diff
@@ -975,26 +993,18 @@ var MSP = {
                     offset += 2;
                     ADVANCED_TUNING.deltaMethod = data.getUint8(offset++, 1);
                     ADVANCED_TUNING.vbatPidCompensation = data.getUint8(offset++, 1);
-                    if (semver.gte(CONFIG.apiVersion, "1.20.0")) {
-                        ADVANCED_TUNING.ptermSetpointWeight = data.getUint8(offset++, 1);
-                        ADVANCED_TUNING.dtermSetpointWeight = data.getUint8(offset++, 1);
-                        ADVANCED_TUNING.toleranceBand = data.getUint8(offset++, 1);
-                        ADVANCED_TUNING.toleranceBandReduction = data.getUint8(offset++, 1);
-                        ADVANCED_TUNING.itermThrottleGain = data.getUint8(offset++, 1);
-                        ADVANCED_TUNING.pidMaxVelocity = data.getUint16(offset, 1);
-                        offset += 2;
-                        ADVANCED_TUNING.pidMaxVelocityYaw = data.getUint16(offset, 1);
-                    } else {
-                        ADVANCED_TUNING.ptermSetpointWeight = 0;
-                        ADVANCED_TUNING.dtermSetpointWeight = 0;
-                        ADVANCED_TUNING.toleranceBand = 0;
-                        ADVANCED_TUNING.toleranceBandReduction = 0;
-                        ADVANCED_TUNING.itermThrottleGain = 0;
-                        ADVANCED_TUNING.pidMaxVelocity = 0;
-                        ADVANCED_TUNING.pidMaxVelocityYaw = 0;
-                    }
                 }
                 break;
+            case MSP_codes.MSP_SPECIAL_PARAMETERS:
+                 var offset = 0;
+                 RC_tuning.rcYawRate = parseFloat((data.getUint8(offset++) / 100).toFixed(2));
+                 if (CONFIG.flightControllerIdentifier == "BTFL" && semver.gte(CONFIG.flightControllerVersion, "2.8.2")) {
```

Comment:
cool, i just took this code from the commit where the special params command was added.

we could take out the BTFL check here, but I it is still used in these places, so probably best to refactor in a different pr:

```
js/msp.js:            if (CONFIG.flightControllerIdentifier == "BTFL" && semver.gte(CONFIG.flightControllerVersion, "2.8.2")) {
js/serial_backend.js:                    if (CONFIG.flightControllerIdentifier === 'BTFL') {
tabs/configuration.js:        if (CONFIG.flightControllerIdentifier == "BTFL" && semver.gte(CONFIG.flightControllerVersion, "2.8.1")) {
tabs/configuration.js:        if (CONFIG.flightControllerIdentifier == "BTFL" && semver.gte(CONFIG.flightControllerVersion, "2.8.2")) {
tabs/configuration.js:        if (CONFIG.flightControllerIdentifier == "BTFL" && semver.gte(CONFIG.flightControllerVersion, "3.0.0")) {
tabs/configuration.js:        if (CONFIG.flightControllerIdentifier == "BTFL" && semver.gte(CONFIG.flightControllerVersion, "3.0.0")) {
tabs/configuration.js:        if (CONFIG.flightControllerIdentifier == "BTFL" && semver.lt(CONFIG.flightControllerVersion, "2.8.1")) {
tabs/configuration.js:        if (CONFIG.flightControllerIdentifier == "BTFL" && semver.lt(CONFIG.flightControllerVersion, "2.8.2")) {
tabs/configuration.js:        if (CONFIG.flightControllerIdentifier != "BTFL" || semver.lt(CONFIG.flightControllerVersion, "3.0.0")) {
tabs/configuration.js:        if (CONFIG.flightControllerIdentifier == "BTFL" && semver.gte(CONFIG.flightControllerVersion, "2.6.0"))  {
tabs/configuration.js:                if (CONFIG.flightControllerIdentifier == "BTFL" && semver.gte(CONFIG.flightControllerVersion, "2.8.1")) {
tabs/configuration.js:                if (CONFIG.flightControllerIdentifier == "BTFL" && semver.lt(CONFIG.flightControllerVersion, "2.8.1")) {
tabs/configuration.js:                if (CONFIG.flightControllerIdentifier == "BTFL" && semver.gte(CONFIG.flightControllerVersion, "2.8.2")) {
tabs/configuration.js:                if (CONFIG.flightControllerIdentifier == "BTFL" && semver.gte(CONFIG.flightControllerVersion, "3.0.0")) {
tabs/pid_tuning.js:        if (CONFIG.flightControllerIdentifier === "BTFL" && semver.gte(CONFIG.flightControllerVersion, "2.8.0")) {
tabs/pid_tuning.js:    if (CONFIG.flightControllerIdentifier !== "BTFL" || semver.lt(CONFIG.flightControllerVersion, "2.8.0")) {
tabs/pid_tuning.js:        if (CONFIG.flightControllerIdentifier !== "BTFL" || semver.lt(CONFIG.flightControllerVersion, "2.8.1")) {
tabs/pid_tuning.js:                if (targetElement.attr('name') === 'rc_rate' && CONFIG.flightControllerIdentifier !== "BTFL" || semver.lt(CONFIG.flightControllerVersion, "2.8.1")) {
tabs/receiver.js:    if (CONFIG.flightControllerIdentifier === 'BTFL' && semver.gte(CONFIG.flightControllerVersion, '2.8.0')) {
tabs/receiver.js:    this.rateCurve = new RateCurve(CONFIG.flightControllerIdentifier !== 'BTFL' || semver.lt(CONFIG.flightControllerVersion, '2.8.0'));
```